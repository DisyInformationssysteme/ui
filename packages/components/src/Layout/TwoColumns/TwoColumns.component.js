import PropTypes from 'prop-types';
import classnames from 'classnames';
import omit from 'lodash/omit';

import Inject from '../../Inject';
import theme from './TwoColumns.module.scss';
import { FloatingDrawer, Tabs } from '@talend/design-system';

/**
 * @param {object} props react props
 * @example
 <TwoColumns name="Hello world"></TwoColumns>
 */
function TwoColumns({ one, drawers, children, tabs, getComponent, ...props }) {
	const containerCSS = classnames('tc-layout-two-columns', theme.container);
	const sidemenuCSS = classnames('tc-layout-two-columns-left', theme.sidemenu);
	const mainCSS = classnames('tc-layout-two-columns-main', theme.main);
	const style = {
		overflow: 'auto',
		height: '100%',
		display: 'flex',
		flexDirection: 'column',
	};
	const safeOne = Inject.getReactElement(getComponent, one);

	return (
		<div className={containerCSS} {...omit(props, 'getComponents')}>
			<div className={sidemenuCSS} id="tc-layout-side-menu" tabIndex="-1">
				{safeOne}
			</div>
			<div className={mainCSS} role="main" id="tc-layout-main" tabIndex="-1">
				<FloatingDrawer.Container>
					{tabs && <Tabs {...tabs} />}
					<div style={style}>{children}</div>
					{drawers.map((drawer, index) => (
						<FloatingDrawer key={index} visible>
							{drawer}
						</FloatingDrawer>
					))}
				</FloatingDrawer.Container>
			</div>
		</div>
	);
}

TwoColumns.displayName = 'TwoColumns';

TwoColumns.propTypes = {
	one: Inject.getReactElement.propTypes,
	children: PropTypes.node,
	drawers: PropTypes.arrayOf(PropTypes.node),
	getComponent: PropTypes.func,
};

export default TwoColumns;
