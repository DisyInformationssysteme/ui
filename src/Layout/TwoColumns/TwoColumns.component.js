import React from 'react';
import theme from './TwoColumns.scss';

/**
 * @param {object} props react props
 * @example
<TwoColumns name="Hello world"></TwoColumns>
 */
function TwoColumns(props) {
	return (
		<div className={theme.container}>
			<div className={theme.sidemenu}>
				{props.one}
			</div>
			<div className={theme.main}>
				{props.two}
			</div>
		</div>
	);
}

TwoColumns.propTypes = {
	one: React.PropTypes.element,
	two: React.PropTypes.element,
};

export default TwoColumns;
