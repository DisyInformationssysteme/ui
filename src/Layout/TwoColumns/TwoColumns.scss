////
/// Layout
/// @group Custom widgets
////

/// Layout first columns in two columns background color
/// @type Color
$tc-layout-two-columns-sidemenu-bg: #4F8992 !default;

.container {
	display: flex;
	flex: 1 0 100%;
	min-height: 0;
}

.sidemenu {
	background-color: $tc-layout-two-columns-sidemenu-bg;

	:global(.nav-pills) :global(.btn-link) {
		color: #FFF;
		width: 100%;
	}

	:global(.btn-link):hover {
		background-color: darken($tc-layout-two-columns-sidemenu-bg, 12%);
	}
}
