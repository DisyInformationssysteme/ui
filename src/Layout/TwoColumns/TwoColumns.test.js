import React from 'react';
import renderer from 'react-test-renderer';

import TwoColumns from './TwoColumns.component';

describe('TwoColumns', () => {
	it('should render its name', () => {
		const wrapper = renderer.create(<TwoColumns name="Hello world" />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
