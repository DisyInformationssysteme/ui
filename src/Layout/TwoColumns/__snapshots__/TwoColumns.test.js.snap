exports[`TwoColumns should render its name 1`] = `
<div
  className={undefined}>
  <div
    className={undefined} />
  <div
    className={undefined} />
</div>
`;
